sonic-net/sonic-buildimage
# Add netconf docker to build-image

#### Why I did it
I wanted to add NETCONF server support for configuring SONiC switches. Similar to the work done on RESTCONF server.

#### How I did it
First, the changes themselves are on the [`Official sonic netconf repositroy`](https://github.com/sonic-net/sonic-netconf-server).
This work focuses on integrating the netconf server in build-image processes. I updated relevant build scripts , dockerfiles and git modules to include the netconf server in the final image.

#### How to verify it
1) Normally build the image
2) Install new image with netconf server on a switch. After switch stabilizes inspect the docker to see the netconf-server docker image is up.
3) Test NETCONF connection using `ssh <username>@<ip> -p 830 -s netconf`



#### Description for the changelog


This change adds support for a RFC compliant NETCONF server for SONiC switches.



Implements sonic-net/SONiC#1466
Implements sonic-net/SONiC#1498
Uses sonic-net/sonic-netconf-server#2
